src/formats/notion-api/database-helpers.ts

```diff
@@ -0,0 +1,864 @@
+/**
+ * Database helper functions for Notion API importer
+ * Handles database conversion and .base file generation
+ */
+
+import { Client, BlockObjectResponse, DatabaseObjectResponse, PageObjectResponse } from '@notionhq/client';
+import { Vault, normalizePath, App } from 'obsidian';
+import { ImportContext } from '../../main';
+import { sanitizeFileName } from '../../util';
+import { getUniqueFolderPath } from './vault-helpers';
+import { makeNotionRequest } from './api-helpers';
+import { canConvertFormula, convertNotionFormulaToObsidian, getNotionFormulaExpression } from './formula-converter';
+import { DatabaseInfo, RelationPlaceholder } from './types';
+import type { FormulaImportStrategy } from '../notion-api';
+
+/**
+ * Obsidian property types that are supported
+ */
+const OBSIDIAN_PROPERTY_TYPES = {
+	CHECKBOX: 'checkbox',
+	DATE: 'date',
+	DATETIME: 'datetime',
+	LIST: 'list',
+	NUMBER: 'number',
+	TEXT: 'text',
+};
+
+/**
+ * Convert a child_database block to Markdown
+ * This creates a reference to a .base file and sets up the database structure
+ * In fact, Notion allows top-level databases, which will be addressed later.
+ */
+export async function convertChildDatabase(
+	block: BlockObjectResponse,
+	ctx: ImportContext,
+	currentPageFolderPath: string,
+	client: Client,
+	vault: Vault,
+	app: App,
+	outputRootPath: string,
+	formulaStrategy: FormulaImportStrategy,
+	processedDatabases: Map<string, DatabaseInfo>,
+	relationPlaceholders: RelationPlaceholder[],
+	importPageCallback: (pageId: string, parentPath: string) => Promise<void>,
+	onPagesDiscovered?: (count: number) => void
+): Promise<string> {
+	if (block.type !== 'child_database') return '';
+	
+	try {
+		// Get database details
+		const databaseId = block.id;
+		const database = await makeNotionRequest(
+			() => client.databases.retrieve({ database_id: databaseId }) as Promise<DatabaseObjectResponse>,
+			ctx
+		);
+		
+		// Extract database title
+		const databaseTitle = extractDatabaseTitle(database);
+		const sanitizedTitle = sanitizeFileName(databaseTitle || 'Untitled Database');
+		
+		ctx.status(`Processing database: ${sanitizedTitle}...`);
+		
+		// In Notion API v2025-09-03, databases have data_sources array
+		// Get the first data source to retrieve properties
+        // It seems that the code I used in the Notion Flow plugin is not compatible with the new 2025-09-03 version of the API.
+        // FIXME: At the same time, I may need to inform the user in the "Import Notes" that the imported database only reads the first data source.
+		
+		let dataSourceProperties: Record<string, any> = {};
+		let propertyIds: string[] = [];
+		
+		if ((database as any).data_sources && (database as any).data_sources.length > 0) {
```

Comment:
Why is `as any` required here?

Reply:
This was an error; the Notion Database API does not return the same property order as the UI, and it has been corrected here.